extensions/spring-xd-extension-http/src/main/java/org/springframework/integration/x/http/NettyHttpInboundChannelAdapter.java

```diff
@@ -286,6 +335,58 @@ public void writeRequested(ChannelHandlerContext ctx, MessageEvent e) throws Exc
 		}
 	}
 
+	/* start whu */
+	void wrapSendMessage(Message<?> message) {
+		sendMessage(message);
+	}
+
+	private boolean sendMessageToResponsibleSender(HttpRequest request, Message<?> message) {
+		String name = "wrapSendMessage";
+		Object sender = getResponsibleSender(request);
+		for (Method m : sender.getClass().getDeclaredMethods()) {
```

Comment:
See my note above about storing the method instead of the "sender" to avoid this each time.

Also, consider using `ReflectionUtils.doWithMethods()` with a method filter instead of this raw iteration.